### PR TITLE
fix: allow rule with empty schedule expression

### PR DESCRIPTION
**Pull request by embano1**
Closes: #1984

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/resource/rule/hooks.go [hunk 1]
```diff
@@ -133,4 +133,21 @@ func customPreCompare(
...
 		delta.Add("Spec.Targets", desired.ko.Spec.Targets, latest.ko.Spec.Targets)
 	}
+
+	// ideally EqualStrings should do but we're missing one case there (see
+	// EqualStrings function comment)
+	desiredExpression := desired.ko.Spec.ScheduleExpression
+	latestExpression := latest.ko.Spec.ScheduleExpression
+	if !(pkgtags.EqualZeroString(desiredExpression) && pkgtags.EqualZeroString(latestExpression)) {
+		delta.Add("Spec.ScheduleExpression", desiredExpression, latestExpression)
+	}
```
Comment (a-hilaly) [inline pkg/resource/rule/hooks.go:143]: Are we missing the case where `*desiredExpression != *latestExpression`?

Comment (embano1) [reply pkg/resource/rule/hooks.go]: We do 😶‍🌫️ (fixed w/ unit tests)

**Review by a-hilaly — APPROVED** [review]
Thank you sir 🙇 !
/lgtm

**Comment by a-hilaly** [comment]
@embano1 time to ship a new release?

**Comment by embano1** [comment]
> @embano1 time to ship a new release?

if we need, sure. or we wait until I fix the related issue?

**Pull request merged**